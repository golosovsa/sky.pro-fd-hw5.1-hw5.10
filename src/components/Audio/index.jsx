import styled from 'styled-components'

const AudioStyled = styled.audio`
  display: none;
`

const Audio = ({ src, forwardRef }) => {
  const handleError = () => {
    throw new Error('loading error')
  }

  return <AudioStyled src={src} ref={forwardRef} onError={handleError} />
}

export default Audio
