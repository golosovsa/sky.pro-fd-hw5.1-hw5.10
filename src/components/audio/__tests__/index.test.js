import { render, screen } from '@testing-library/react'
import Audio from '..'

describe('<Audio />', () => {
  describe('Snapshot test', () => {
    it('should render successfully and match snapshot', () => {
      const { container } = render(<Audio />)

      expect(container).toMatchSnapshot()
    })
  })

  describe('style test', () => {
    it('should have display: none', () => {
      render(<Audio />)

      expect(screen.getByRole('Audio', { hidden: true }))
    })
  })
})
