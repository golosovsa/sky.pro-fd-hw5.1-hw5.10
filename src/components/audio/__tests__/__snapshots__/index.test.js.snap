// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Audio /> Snapshot test should render successfully and match snapshot 1`] = `
<div>
  <audio
    class="sc-bcXHqe xuMZq"
  />
</div>
`;

exports[`<audio /> should render successfully and match snapshot 1`] = `
<div>
  <audio
    class="sc-bcXHqe xuMZq"
  />
</div>
`;
